src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionCertificateStoreProvider.java

```diff
@@ -74,11 +101,123 @@ public String getName() {
+        if (certParts.length > 3) {
+            throw new SQLServerException(SQLServerException.getErrString("R_AECertpathBad"), null);
+        }
+
+        // Extract the store location where the cert is stored
+        if (certParts.length > 2) {
+            if (certParts[0].equalsIgnoreCase(localMachineDirectory)) {
+                storeLocation = localMachineDirectory;
+            }
+            else if (certParts[0].equalsIgnoreCase(currentUserDirectory)) {
+                storeLocation = currentUserDirectory;
+            }
+            else {
+                throw new SQLServerException(SQLServerException.getErrString("R_AECertLocBad"), null);
+            }
+        }
+
+        // Parse the certificate store name. Only store name "My" is supported.
+        if (certParts.length > 1) {
+            if (!certParts[certParts.length - 2].equalsIgnoreCase(myCertificateStore)) {
+                throw new SQLServerException(SQLServerException.getErrString("R_AECertLocBad"), null);
+            }
+        }
+
+        // Get thumpbrint
+        String thumbprint = certParts[certParts.length - 1];
+        if ((null == thumbprint) || (0 == thumbprint.length())) {
+            // An empty thumbprint specified
+            throw new SQLServerException(SQLServerException.getErrString("R_AECertHashEmpty"), null);
+        }
+
+        // Find the certificate and return
+        return getCertificateByThumbprintNonWin(thumbprint, masterKeyPath);
+    }
+
+    private String getThumbPrint(X509Certificate cert) throws NoSuchAlgorithmException, CertificateEncodingException {
+        MessageDigest md = MessageDigest.getInstance("SHA-1");
+        byte[] der = cert.getEncoded();
+        md.update(der);
+        byte[] digest = md.digest();
+        return DatatypeConverter.printHexBinary(digest);
+    }
+
+    private CertificateDetails getCertificateByThumbprintNonWin(String thumbprint,
+                                                                String masterKeyPath) throws SQLServerException {
+        FileInputStream fis;
+
+        if (null == masterKeyPath || 0 == masterKeyPath.length()) {
+            throw new SQLServerException(null, SQLServerException.getErrString("R_InvalidMasterKeyDetails"), null, 0, false);
+        }
+
+        // Loading key store as PKCS12
+        KeyStore keyStore = null;
+        try {
+            keyStore = KeyStore.getInstance("PKCS12");
+            fis = new FileInputStream(keyStorePath);
+            keyStore.load(fis, keyStorePwd);
+        } catch (IOException | CertificateException | KeyStoreException | NoSuchAlgorithmException e) {
+            throw new SQLServerException(SQLServerException.getErrString("R_invalidKeyStoreFile"), e);
+        }
+
+        // If we are here, we were able to load a PKCS12 file.
+        try {
+            for (Enumeration<String> enumeration = keyStore.aliases(); enumeration.hasMoreElements();) {

```

Comment:
**Reporter**: CHECKSTYLE
**Rule**: com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck
**Severity**: ERROR
**File**: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionCertificateStoreProvider.java L194

Line is longer than 80 characters (found 104).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<1301146174>*